string type
string initialInfo
---
string output
